ddtrace/internal/telemetry/metrics_namespaces.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any
+from typing import Dict
+
+from ddtrace.internal.constants import TELEMETRY_APPSEC
+from ddtrace.internal.constants import TELEMETRY_TRACER
+from ddtrace.internal.telemetry.metrics import CountMetric
+from ddtrace.internal.telemetry.metrics import GaugeMetric
+from ddtrace.internal.telemetry.metrics import Metric
+from ddtrace.internal.telemetry.metrics import MetricTagType
+from ddtrace.internal.telemetry.metrics import MetricType
+from ddtrace.internal.telemetry.metrics import RateMetric
+
+
+NamespaceMetricType = Dict[str, Dict[str, Any]]
+
+
+class TelemetryTypeError(Exception):
+    pass
+
+
+class MetricNamespace:
+    _metrics_data = {TELEMETRY_TRACER: {}, TELEMETRY_APPSEC: {}}  # type: NamespaceMetricType
+    metric_class = {"count": CountMetric, "gauge": GaugeMetric, "rate": RateMetric}
+
+    def _flush(self):
+        self._metrics_data = {TELEMETRY_TRACER: {}, TELEMETRY_APPSEC: {}}
+
+    def get(self):
+        # type: () -> Dict[str, Any]
+        namespace_metrics = self._metrics_data.copy()
+        return namespace_metrics
+
+    def validate_type_metric(self, metric_check, metric_to_validate):
+        # type: (Metric, Metric) -> bool
+        if metric_check.__class__ == metric_to_validate.__class__:
```

Comment:
why checking only the class ?

Reply:
ok, but the way it is done here is based on id, not name. So two metrics with the same name but different tags will still be possible. is this ok?

Reply:
yes, two metrics with the same name but different tags it will be reported as two different metrics (check it https://www.datadoghq.com/blog/the-power-of-tagged-metrics/#whats-a-metric-tag)

But we CAN NOT have two metrics with the same name and different type

Reply:
ok, this looks perfect!